troubadix/plugins/security_messages.py

```diff
@@ -14,34 +14,70 @@
 #
 #  You should have received a copy of the GNU General Public License
 #  along with this program.  If not, see <http://www.gnu.org/licenses/>.
-import re
 from pathlib import Path
 from typing import Iterator
 
 from troubadix.helper.patterns import ScriptTag, get_script_tag_pattern
 from troubadix.plugin import FileContentPlugin, LinterError, LinterResult
 
+SECURITY_MESSAGE_IMPLEMENTATIONS = [
+    "security_message",
+    "http_check_remote_code",
+    "citrix_xenserver_check_report_is_vulnerable",
+    "citrix_xenserver_report_missing_patch",
+]
+
+
+def _file_contains_security_message(file_content: str) -> bool:
+    """Checks wether a VT content contains a call to security_message
+    or any function known to implement it
+
+    Args:
+        file_content (str): The content of the VT
+    """
+    return any(
+        security_message in file_content
+        for security_message in SECURITY_MESSAGE_IMPLEMENTATIONS
+    )
+
 
 class CheckSecurityMessages(FileContentPlugin):
     name = "check_security_messages"
 
-    def check_content(
-        self,
-        nasl_file: Path,
-        file_content: str,
+    def _check_security_message_present(
+        self, nasl_file: Path, file_content: str
     ) -> Iterator[LinterResult]:
-        """This script checks the passed VT if it is using a security_message
-        and having no severity (CVSS score) assigned which is an error /
-        debugging leftover in most cases.
+        """Checks that the VT does have a
+        security_message or implementing function call
 
         Args:
-            nasl_file: The VT that is going to be checked
-            file_content: The content of the VT
-
+            nasl_file (Path): The VTs path
+            file_content (str): The content of the VT
         """
-        if nasl_file.suffix == ".inc":
+        deprecated_pattern = get_script_tag_pattern(
+            script_tag=ScriptTag.DEPRECATED
+        )
+        if deprecated_pattern.match(file_content):
             return
```

Comment:
It seems this *might* not work as expected (e.g. a problem in the helper function) because in e.g. greenbone/vulnerability-tests#4117 i'm seeing a false positive for a deprecated VT having this tag:

```
  script_tag(name:"deprecated", value:TRUE);
```

Reply:
On it

Reply:
```suggestion
        deprecated_pattern = get_script_tag_pattern(
            script_tag=ScriptTag.DEPRECATED
        )
        if deprecated_pattern.search(file_content):
            return
```

Seems this is working, at least for me.